File: AdMixerSDKSample/app/src/main/java/com/nasmedia/admixer/sample/RewardInterstitialVideoActivity.java
package com.nasmedia.admixer.sample;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.nasmedia.admixer.ads.AdEvent;
import com.nasmedia.admixer.ads.AdInfo;
import com.nasmedia.admixer.ads.AdListener;
import com.nasmedia.admixer.ads.RewardInterstitialVideoAd;

import java.util.HashMap;
import java.util.Map;

public class RewardInterstitialVideoActivity extends AppCompatActivity {

    private Button btnRewardVideoShow;
    private ProgressBar progressBar;
    private RewardInterstitialVideoAd rewardInterstitialVideoAd;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rewardvideo);

        progressBar = findViewById(R.id.loading_bar);
        btnRewardVideoShow = findViewById(R.id.btn_reward_video_show);

        Map<String, String> params = new HashMap<>();
        params.put("use_id", "nas");
        params.put("name", "choi");
        params.put("phone", "010-1111-1111");

        AdInfo adInfo = new AdInfo.Builder(Application.ADUNIT_ID_REWARD_INTERSTITIAL_VIDEO) // AdMixer 플랫폼에서 발급받은 리워드 전면 비디오 ADUNIT_ID
                .interstitialTimeout(0) // 초단위로 전면 광고 타임아웃 설정 (기본값 : 0, 0 이면 서버지정 시간으로 처리, 서버지정 시간 : 20s)
                .maxRetryCountInSlot(-1) // 리로드 시간 내에 반복 횟수(-1 : 무한, 0 : 반복 없음, n : n번 반복)
                .isRetry(true) // 광고 재요청 설정 (true - 기본값), false 시, 1회 요청 후 바로 Callback
                .setCustomParams(params) // Reward Callback 커스텀데이터 Map형태로 추가 (선택사항)
                .build();

        rewardInterstitialVideoAd = new RewardInterstitialVideoAd(this);
        // 이 때 설정하신 RewardInterstitialVideoAd 의 부모 activity 는 원활한 광고 제공을 위해 hardwareAccelerated 가 true 설정되오니 참고 부탁드립니다.
        rewardInterstitialVideoAd.setAdInfo(adInfo);
        rewardInterstitialVideoAd.setListener(new AdListener() {
            @Override
            public void onReceivedAd(Object o) {
                // 광고 수신 성공
                Toast.makeText(RewardInterstitialVideoActivity.this, "onReceivedAd", Toast.LENGTH_SHORT).show();
                btnRewardVideoShow.setText("리워드 전면 비디오 광고보기");
                progressBar.setVisibility(View.GONE);
            }

            @Override
            public void onFailedToReceiveAd(Object o, int i, String s) {
                // 광고 수신 실패
                Toast.makeText(RewardInterstitialVideoActivity.this, s, Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onEventAd(Object o, AdEvent adEvent) {
                // 기타 이벤트
                switch (adEvent) {
                    case CLOSE:   // 광고 창이 닫혔을 때
                    case SKIPPED: // 동영상 SKIP 버튼 클릭 시
                        rewardInterstitialVideoAd.closeRewardVideoAd();
                        btnRewardVideoShow.setText("리워드 전면 비디오 광고요청");
                        break;
                    case COMPLETION: // 동영상 재생 완료 시
                    case CLICK: // 더보기 버튼 클릭 시
                        break;
                }
            }
        });

        // 광고 요청 후 광고를 받은 뒤, 원하는 시점에 노출
        btnRewardVideoShow.setOnClickListener(v -> {
            if (rewardInterstitialVideoAd.hasInterstitial)
                rewardInterstitialVideoAd.showRewardVideoAd(); // 광고를 노출한다.
            else {

                rewardInterstitialVideoAd.loadRewardVideoAd(); // 광고를 미리 로드한다.
                progressBar.setVisibility(View.VISIBLE);
            }
        });
    }

    // 생명주기에 따라 아래 설정이 반드시 필요합니다.
    @Override
    protected void onDestroy() {
        if (rewardInterstitialVideoAd != null) {
            rewardInterstitialVideoAd.stopRewardVideoAd();
            rewardInterstitialVideoAd = null;
        }
        super.onDestroy();
    }
}

File: AdMixerSDKKotlinSample/app/src/main/java/com/nasmedia/admixerkotlin/RewardInterstitialVideoActivity.kt
package com.nasmedia.admixerkotlin

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.nasmedia.admixer.ads.AdEvent
import com.nasmedia.admixer.ads.AdInfo
import com.nasmedia.admixer.ads.AdListener
import com.nasmedia.admixer.ads.RewardInterstitialVideoAd

class RewardInterstitialVideoActivity : AppCompatActivity() {
    private lateinit var btnRewardVideoShow: Button
    private lateinit var progressBar: ProgressBar
    private var rewardInterstitialVideoAd: RewardInterstitialVideoAd? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rewardvideo)
        progressBar = findViewById(R.id.loading_bar)
        btnRewardVideoShow = findViewById(R.id.btn_reward_video_show)

        val params: MutableMap<String, String> = HashMap()
        params["use_id"] = "nas"
        params["name"] = "choi"
        params["phone"] = "010-1111-1111"

        val adInfo = AdInfo.Builder(Application.ADUNIT_ID_REWARD_INTERSTITIAL_VIDEO) // AdMixer 플랫폼에서 발급받은 리워드 전면 비디오 ADUNIT_ID
                .interstitialTimeout(0) // 초단위로 전면 광고 타임아웃 설정 (기본값 : 0, 0 이면 서버지정 시간으로 처리, 서버지정 시간 : 20s)
                .maxRetryCountInSlot(-1) // 리로드 시간 내에 반복 횟수(-1 : 무한, 0 : 반복 없음, n : n번 반복)
                .isRetry(true) // 광고 재요청 설정 (true - 기본값), false 시, 1회 요청 후 바로 Callback
                .setCustomParams(params) // Reward Callback 커스텀데이터 Map형태로 추가 (선택사항)
                .build()
        rewardInterstitialVideoAd = RewardInterstitialVideoAd(this)
        // 이 때 설정하신 RewardInterstitialVideoAd 의 부모 activity 는 원활한 광고 제공을 위해 hardwareAccelerated 가 true 설정되오니 참고 부탁드립니다.
        rewardInterstitialVideoAd!!.setAdInfo(adInfo)
        rewardInterstitialVideoAd!!.setListener(object : AdListener {
            override fun onReceivedAd(o: Any) {
                // 광고 수신 성공
                Toast.makeText(
                    this@RewardInterstitialVideoActivity,
                    "onReceivedAd",
                    Toast.LENGTH_SHORT
                ).show()
                btnRewardVideoShow.setText("리워드 전면 비디오 광고보기")
                progressBar.setVisibility(View.GONE)
            }

            override fun onFailedToReceiveAd(o: Any, i: Int, s: String) {
                // 광고 수신 실패
                Toast.makeText(this@RewardInterstitialVideoActivity, s, Toast.LENGTH_SHORT).show()
            }

            override fun onEventAd(o: Any, adEvent: AdEvent) {
                // 기타 이벤트
                when (adEvent) {
                    AdEvent.CLOSE, AdEvent.SKIPPED -> { // 광고 창이 닫혔을 때
                        rewardInterstitialVideoAd!!.closeRewardVideoAd()
                        btnRewardVideoShow.setText("리워드 전면 비디오 광고요청")
                    }
                    AdEvent.DISPLAYED -> {} // 광고 노출
                    else -> {}
                }
            }
        })

        // 광고 요청 후 광고를 받은 뒤, 원하는 시점에 노출
        btnRewardVideoShow.setOnClickListener(View.OnClickListener { v: View? ->
            if (rewardInterstitialVideoAd!!.hasInterstitial) rewardInterstitialVideoAd!!.showRewardVideoAd() // 광고를 노출한다.
            else {
                rewardInterstitialVideoAd!!.loadRewardVideoAd() // 광고를 미리 로드한다.
                progressBar.setVisibility(View.VISIBLE)
            }
        })
    }

    // 생명주기에 따라 아래 설정이 반드시 필요합니다.
    override fun onDestroy() {
        if (rewardInterstitialVideoAd != null) {
            rewardInterstitialVideoAd!!.stopRewardVideoAd()
            rewardInterstitialVideoAd = null
        }
        super.onDestroy()
    }
}